crates/bevy_ecs/src/event/trigger.rs

```diff
@@ -0,0 +1,461 @@
+    /// - `observers` must correspond to observers compatible with the event type `E`
+    /// - Read and abide by the "Safety" section defined in the top-level [`Trigger`] docs. Calling this function is
+    ///   unintuitively risky. _Do not use it directly unless you know what you are doing_. Importantly, this should only
+    ///   be called for an `event` whose [`Event::Trigger`] matches this trigger.
+    unsafe fn trigger(
+        &mut self,
+        world: DeferredWorld,
+        observers: &CachedObservers,
+        trigger_context: &TriggerContext,
+        event: &mut E,
+    );
+}
+
+/// A [`Trigger`] that runs _every_ "global" [`Observer`](crate::observer::Observer) (ex: registered via [`World::add_observer`](crate::world::World::add_observer))
+/// that matches the given [`Event`].
+///
+/// The [`Event`] derive defaults to using this [`Trigger`], and it is usable for any [`Event`] type.
+#[derive(Default)]
+pub struct GlobalTrigger;
+
+// SAFETY:
+// - `E`'s [`Event::Trigger`] is constrained to [`GlobalTrigger`]
+// - The implementation abides by the other safety constraints defined in [`Trigger`]
+unsafe impl<E: for<'a> Event<Trigger<'a> = Self>> Trigger<E> for GlobalTrigger {
+    unsafe fn trigger(
+        &mut self,
+        world: DeferredWorld,
+        observers: &CachedObservers,
+        trigger_context: &TriggerContext,
+        event: &mut E,
+    ) {
+        self.trigger_internal(world, observers, trigger_context, event.into());
+    }
+}
+
+impl GlobalTrigger {
+    /// # Safety
+    /// - `observers` must come from the `world` [`DeferredWorld`], and correspond to observers that match the `event` type
+    /// - `event` must point to an [`Event`]
+    /// -  The `event` [`Event::Trigger`] must be [`GlobalTrigger`]
+    /// - `trigger_context`'s [`TriggerContext::event_key`] must correspond to the `event` type.
+    /// - Read, understand, and abide by the [`Trigger`] safety documentation
+    unsafe fn trigger_internal(
+        &mut self,
+        mut world: DeferredWorld,
+        observers: &CachedObservers,
+        trigger_context: &TriggerContext,
+        mut event: PtrMut,
+    ) {
+        // SAFETY: `observers` is the only active reference to something in `world`
+        unsafe {
+            world.as_unsafe_world_cell().increment_trigger_id();
+        }
+        for (observer, runner) in observers.global_observers() {
+            // SAFETY:
+            // - `observers` come from `world` and match the `event` type, enforced by the call to `trigger_internal`
+            // - the passed in event pointer is an `Event`, enforced by the call to `trigger_internal`
+            // - `trigger` is a matching trigger type, as it comes from `self`, which is the Trigger for `event`, enforced by `trigger_internal`
+            // - `trigger_context`'s event_key matches `E`, enforced by the call to `trigger_internal`
+            // - this abides by the nuances defined in the `Trigger` safety docs
```

Comment:
(In relation to my previous comment)

This is the kind of "pinky promise" I was talking about and doesn't really prove anything IMO.

In this case the trigger does not hold lifetimes so there's trivially no issue with them, but in other cases that's not true.